webui/react/src/components/Dropdown.tsx

```diff
@@ -121,17 +121,19 @@ const Dropdown: React.FC<Props> = ({
   if (showArrow) classes.push(css.arrow);
 
   const handleClick = useCallback(
-    (event: any) => {
+    (event: Event) => {
       if (!event || !event.target) return;
 
       event.stopPropagation();
 
+      const target = event.target as Element;
       const isAntPicker =
-        event.target.closest('div') &&
-        typeof event.target.closest('div').className === 'string' &&
-        event.target.closest('div').className.indexOf('ant-picker') >= 0;
-      const isTrigger = triggerRef.current?.contains(event.target);
-      const isDropdown = dropdownRef.current?.contains(event.target);
+        (target.closest('div') &&
+          typeof target.closest('div')?.className === 'string' &&
+          target.closest('div')?.className?.indexOf('ant-picker')) ||
+        -1 >= 0;
```

Comment:
Whats this?

Reply:
I think this means Dropdown should behave differently if it's used right within ant-picker.
But upon a quick look through the code base, I don't think this happens anymore.

Reply:
i think `-1` should be inside parentheses?

Reply:
Things in parentheses may be undefined, that's way we need the -1 for the comparison

Reply:
current code
```ts
(target.closest('div') && 
typeof target.closest('div')?.className === 'string' && 
target.closest('div')?.className?.indexOf('ant-picker')) ?? -1 >= 0;
```

what i suggest

```ts
```ts
(target.closest('div') && 
typeof target.closest('div')?.className === 'string' && 
(target.closest('div')?.className?.indexOf('ant-picker') ?? -1) >= 0);
```
```

Reply:
You are right, but these should have the same effect?

Reply:
i think so. 
The original code is like this

```ts
event.target.closest('div') &&
        typeof event.target.closest('div').className === 'string' &&
        event.target.closest('div').className.indexOf('ant-picker') >= 0;
```
i think you added `?` to catch `undefined`. if it doesnt exist, `indexOf` returns `-1`, so it should be the same effect.

Reply:
Comparing `undefined` to a number throws an error, that why I added -1

Reply:
yea i know. thats why we need `?` and `?? -1`